### PR TITLE
jar_index cleanout

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 1]
```diff
@@ -227,55 +228,42 @@ java_class_loadert::get_parse_tree(
...
-  auto existing_it = jars_by_path.find(jar_path);
-  if(existing_it != jars_by_path.end())
-    return std::cref(existing_it->second);
-
   std::vector<std::string> filenames;
+
   try
   {
-    filenames = jar_pool(jar_path).filenames();
+    filenames=jar_pool(jar_path).filenames();
```
Comment (mgudemann) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: no change needed, original code corresponds to coding standard

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 2]
```diff
@@ -227,55 +228,42 @@ java_class_loadert::get_parse_tree(
...
-  jar_files.push_front(jar_path);
-
-  for(const auto &e : jar_index->get())
-    operator()(e.first);
-
-  jar_files.pop_front();
-}
-
-java_class_loadert::jar_index_optcreft java_class_loadert::read_jar_file(
+std::vector<irep_idt> java_class_loadert::load_entire_jar(
```
Comment (mgudemann) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: needs documentation

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 3]
```diff
@@ -71,21 +71,28 @@ java_class_loadert::parse_tree_with_overlayst &java_class_loadert::operator()(
...
+  optionalt<std::string> data;
 
-  auto data = jar_pool(jar_file).get_entry(jar_index_it->second);
+  try
+  {
+    data = jar_pool(jar_file).get_entry(filename);
+  }
+  catch(const std::runtime_error &)
+  {
+    error() << "failed to open JAR file `" << jar_file << "'" << eom;
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: Report exn.what()?

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 4]
```diff
@@ -71,21 +71,28 @@ java_class_loadert::parse_tree_with_overlayst &java_class_loadert::operator()(
...
+    return {};
+  }
 
   if(!data.has_value())
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: Clarify: can this ever happen if get_entry returns without an exception?

**Review by smowton — APPROVED** [review]
Couple of nitpicks

File: jbmc/src/java_bytecode/java_class_loader.cpp (hunk 3 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_class_loader.cpp]: We haven't done that yet anywhere -- before doing that, I'd like to first check whether these are useful to the user on our 3 main platforms.

File: jbmc/src/java_bytecode/java_class_loader.cpp (hunk 4 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_class_loader.cpp]: Will add comment.

**Pull request merged**